adagrams/game.py

```diff
@@ -1,11 +1,169 @@
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
 def draw_letters():
-    pass
+    alphabet = list(LETTER_POOL.keys())
+    drawn_letters = []
+
+    while True:
+        # Ensure return list does not have more than 10 elems
+        if len(drawn_letters) == 10:
+            break
+
+        random_letter = alphabet[random.randint(0, len(alphabet)-1)]
+        quantity = LETTER_POOL[random_letter]
+        counter = 0
+        
+        # Letter should not be in return list more than its quantity in the pool
+        for i in range(len(drawn_letters)):
+            if drawn_letters[i] == random_letter:
+                counter += 1
+        if counter < quantity:
+            drawn_letters.append(random_letter)
+
+    return drawn_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    available_letters = copy.deepcopy(letter_bank)
+    word = word.upper()
+
+    # Remove letters from available letters as they are used
+    for letter in word:
+        if letter in available_letters:
+            available_letters.remove(letter)
+        
+        # If the letter isn't in available letters, the word is invalid
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    word = word.upper()
+    score = 0
+
+    # Retrieve points for each letter
+    for letter in word:
+        score += LETTER_VALUE[letter]
+    
+    # Check for word length bonus 
+    # (words longer than 6 letters get an additional 8 points)
+    if len(word) > 6 and len(word) < 11:
+        score += 8
+    
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    words_scores = {}
+    score_counter = 0
+    potential_winners = []
+    winning_word = "blank"
+    word_length = 11
+    shortest_words = []
+    ten_letter_words = []
+
+    # Calculate the scores of each word in the list, 
+    # highest score becomes "score_counter"
+    for word in word_list:
+        words_scores[word] = score_word(word)
+        if words_scores[word] > score_counter:
+            score_counter = words_scores[word]
+
+    # Add any words with the highest score to potential winners list
+    for word in words_scores:
+        if words_scores[word] == score_counter:
+            potential_winners.append(word)
+
+    # If there is only one word in potential winners list then it is the best word!
+    if len(potential_winners) == 1:
+        winning_word = potential_winners[0]
+    
+    # Tiebreaker! 
+    # The winning word is the shortest, unless there is a 10 letter word.
+    else:
+        # Identify if any words are 10 in length
+        # Identify lowest word length in a for loop
+        for word in potential_winners:
+            if len(word) == 10:
+                ten_letter_words.append(word)
+            elif len(word) < word_length:
```

Comment:
Seeing how you use this variable, I wonder if a variable name like `shortest_word_len` would help, although `word_length` is self-explanatory enough.